src/SocketManager.cpp

```diff
@@ -165,24 +206,30 @@ bool SocketManager::readClientData(int fd) {
 
 /* ---------------------------- Handle Responses ---------------------------- */
 
-void SocketManager::processRequestAndRespond(int fd) {
+void SocketManager::processRequest(int fd) {
     HTTPRequest request(this->clientStates[fd].readBuffer);
     const ServerConfig& serverConfig = getCurrentServer(request);
 
 	try {
 		HTTPResponse response;
 		response.prepareResponse(request, serverConfig);
+		clientStates[fd].responseComplete = false;
 		clientStates[fd].writeBuffer = response.convertToString();
-    	sendResponse(fd);
+		// DEBUG("Prepared Response: " << std::endl << clientStates[fd].writeBuffer << std::endl);
+    	// sendResponse(fd);
 	} catch (const std::runtime_error& e) {
 		ERROR(e.what());
 	}
 }
 
-void SocketManager::sendResponse(int fd) {
+void SocketManager::sendResponse(int i) {
+	int fd = this->fds[i].fd;
```

Comment:
Remove this and use the pass down value instead. No need to create a local variable for this